src/test/java/com/neo4j/docker/TestSelfSignedCerts.java

```diff
@@ -0,0 +1,173 @@
+package com.neo4j.docker;
+
+import com.neo4j.docker.utils.Neo4jVersion;
+import com.neo4j.docker.utils.SetContainerUser;
+import com.neo4j.docker.utils.TestSettings;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Neo4jContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.stream.Stream;
+
+import org.neo4j.driver.AuthTokens;
+import org.neo4j.driver.Config;
+import org.neo4j.driver.Driver;
+import org.neo4j.driver.GraphDatabase;
+import org.neo4j.driver.Result;
+import org.neo4j.driver.Session;
+
+public class TestSelfSignedCerts
+{
+    private static Logger log = LoggerFactory.getLogger( TestSelfSignedCerts.class );
+
+    static Stream<Arguments> defaultUserFlag()
+    {
+        // "asUser={0}, secureFlag={1}"
+        // expected behaviour is that if you set --user flag, your data should be read/writable
+        // if you don't set --user flag then read/writability should be controlled by the secure file permissions flag
+        // the asUser=true, secureflag=false combination is tested separately because the container should fail to start.
+        return Stream.of( Arguments.arguments( false ), Arguments.arguments( true ) );
+    }
+
+    private Neo4jContainer setupBasicContainer( boolean asCurrentUser )
+    {
+        log.info( "Running as user {}, {}", asCurrentUser ? "non-root" : "root" );
+
+        Neo4jContainer container = new Neo4jContainer( TestSettings.IMAGE_ID );
+        container.withExposedPorts( 7473, 7474, 7687 )
+                .withLogConsumer( new Slf4jLogConsumer( log ) )
+                .withEnv( "SSL_GENERATE_CERTIFICATES", "yes" )
+                .withEnv( "NEO4J_ACCEPT_LICENSE_AGREEMENT", "yes" );
+
+        if ( asCurrentUser )
+        {
+            SetContainerUser.nonRootUser( container );
+        }
+        return container;
+    }
+
+    @ParameterizedTest( name = "asUser={0}" )
+    @MethodSource( "defaultUserFlag" )
```

Comment:
In TestPasswords.java the same thing is achieve by doing `@ValueSource(booleans = {true, false})` no need for an extra method with junit stuff in then.